addons/udes_stock/controllers/stock_picking_batch.py

```diff
@@ -201,7 +203,20 @@ def reserve_pallet(self, ident, pallet_name):
             raise ValidationError(
                 _("The specified batch is not assigned to you."))
 
-        batch.reserve_pallet(pallet_name)
+        # TODO this is messy
```

Comment:
Everywhere we assume the parameter is the type we expect, so you can remove the try except and go for: 
` pick = StockPicking.browse(int(picking_id))`

Reply:
nit: `picking = XXX`